samples/drivers/soc_flash_stm32h7/README.rst

```diff
@@ -0,0 +1,81 @@
+.. zephyr:code-sample:: soc-flash-stm32h7
```

Comment:
I don't think such sample should be created.
I'm not sure why the _nrf one was created initially, but this isn't a great way to test functionality. If we need to add driver dedicated samples for each new series, that would be a very bad indicator on Zephyr API abstraction capability.
You should be able to use tests/drivers/flash/common for your need

Reply:
@erwango I do see value in the sample. I looked for an example on how to define partitions on an internal flash and use flash API to erase/write/read to the partition. I saw the NVS example, but that is for small flash sizes and doesn't support 128K erase block size. There are so many driver examples under https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/drivers and I am wondering why you don't like the idea, I wanted to run an example to do this from M7 and M4 and found the soc_flash_nrf example which show case such an example.  Perhaps https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/drivers/soc_flash_nrf can be renamed to soc_flash and I will investigate if I can extend this to STM32H747 as well. Does that make sense to you? 

@de-nordic Do you think I can rename soc_flash_nrf to soc_flash and add support for STM32H747i-disco? I will investigate to see if it is generic enough to work on STM32H747i-disco. Perhaps I have to add some #ifdef around some part of the code if it is not applicable for STM32H747i-disco.